api4/insights.go

```diff
@@ -340,6 +344,106 @@ func getTopThreadsForUserSince(c *Context, w http.ResponseWriter, r *http.Reques
 	w.Write(js)
 }
 
+// Top Channels
+
+func getTopInactiveChannelsForTeamSince(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	team, err := c.App.GetTeam(c.Params.TeamId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), team.Id, model.PermissionViewTeam) {
+		c.SetPermissionError(model.PermissionViewTeam)
+		return
+	}
+
+	user, err := c.App.GetUser(c.AppContext.Session().UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	loc := user.GetTimezoneLocation()
+	startTime := model.StartOfDayForTimeRange(c.Params.TimeRange, loc)
+
+	topChannels, err := c.App.GetTopInactiveChannelsForTeamSince(c.AppContext, c.Params.TeamId, c.AppContext.Session().UserId, &model.InsightsOpts{
+		StartUnixMilli: startTime.UnixMilli(),
+		Page:           c.Params.Page,
+		PerPage:        c.Params.PerPage,
+	})
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	js, jsonErr := json.Marshal(topChannels)
+	if jsonErr != nil {
+		c.Err = model.NewAppError("getTopInactiveChannelsForTeamSince", "api.marshal_error", nil, jsonErr.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Write(js)
```

Comment:
Hey Shivasish, apologies for the drive-by comment. Just one small thing - usually we follow the convention:

```go
if err := json.NewEncoder(w).Encode(topChannels); err != nil {
  c.Logger.Warn("Error while writing top channels", mlog.Err(err))
}
```

This does the marshalling and writing the response in a single line, and also captures the error from writing the bytes to the network.

Reply:
Thanks for the suggestion, this seems a lot better. I'll apply this change to wherever we're doing this separately.

Reply:
Fixed, thanks! Also added a ticket to fix it in other endpoints, https://mattermost.atlassian.net/browse/MM-46331